### PR TITLE
Trying a thing

### PR DESCRIPTION
**Pull request by NCrusher74**
No matter how I format the paths in `headerSearchPath`, I can't seem to find what it needs to locate the files.

This build works fine when imported as a framework. I just want to make a package out of it so changes to the project I'm using it in won't cause difficulties.

### CODE REVIEW

**Comment by SDGGiesbrecht** [comment]
Since you asked for review, you moved the repository, marking it as `-old`. The links in the messages GitHub sent me were dead by the time I got to them. I just now discovered it was moved, not deleted. Do you still want me to look at this?

**Comment by NCrusher74** [comment]
I found a different wrapper for TagLib, which is why I renamed it because
it wanted TagLib compiled as a framework instead of the files. I’m not sure
that will work as a package now? I saw something last night about SPM not
working with binary frameworks?

However it would be nice to know if I was getting anywhere near close with
what I was trying to do with this one.

**Comment by NCrusher74** [comment]
Sorry I just got home and now that I'm not trying to type on my phone I realized that what I said probably didn't make much sense.

Basically, what happened yesterday is that, since I didn't want to pester you, I posted to the TagLib mailing list seeing if anyone had any advice on fixing a few of the issues I was having with the wrapper I had cobbled together from the IOS wrapper and the version of TagLib I had found.

I don't know why I never thought to ask the mailing list before now, except that I assumed mailing lists were a thing we all left behind in the late 90s? But if I had, I could have saved us both months of headaches because it turned out someone on the list there had a wrapper that, in theory, should do most of what I'm hoping to accomplish. I haven't put it through its paces yet because I'd still much rather deal with `ID3TagEditor` because it comes nicely packaged in SPM and we've put so much effort into making it work, but seeing if I could find a way to work with TagLib is something I wanted to try out anyway because of all the frustration it's caused me, having a tool *right there* that does everything I need but wasn't accessible.

Anyway, I digress.

So, I renamed the TagLibKit I had been working on and tucked it aside because the new wrapper expects the binary framework build of TagLib so if I wanted to go on experimenting with making `TagLibKit`/`SwitTagger`, it would be easier to start a project from scratch than change the existing project.

But the other side of the `TagLibKit` experiment was packaging up nicely in Swift Package Manager, in order to learn how to use SPM (which is something I'll need to do for `AudiobookTagger`, eventually) and that I haven't accomplished yet. I'm not sure I can with this new wrapper that expects me to use the `tag.framework` build instead of files themselves, because I came across something yesterday that said you can't use binary frameworks in an SPM package,  but maybe I can at least use the pieces of code that deal with the functions the other wrappers lacked to piece together wrappers that will work with the `TagLibKit-old` file structure while having the features that version is lacking.

So what I'm saying is, if and when you have time, I'd really like to know if I was anywhere near close to being on the mark with my efforts to create an SPM package with this version, and if not, what I was getting wrong. Because that is something I need to learn and I just can't seem to get it right. And since doing it for `AudiobookTagger` is going to be complicated due to using `MP42Foundation`, I need to have that trick in my bag.

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        // .package(url: /* package url */, from: "1.0.0"),
+    ],
+    targets: [
+        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
+        // Targets can depend on other targets in this package, and on products in packages which this package depends on.
+        .target(
+            name: "TagLib",
+            path: "Sources/TagLib",
+            cxxSettings: [
+            .headerSearchPath("../taglib-config"),
```
[the remark this answers is not in the document]
Comment (SDGGiesbrecht) [reply Package.swift]: They are relative to the `path` two lines earlier.

If you had left all the files where they were, and merely added a manifest, then the code I posted would have worked verbatim (except for the elipses). I actually did build it that way.

With the files in their new locations, this entry would be just `.headerSearchPath("taglib-config")`, but I’m not sure what other problems may have been caused by shuffling the files around.

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            ]
+        ),
+        .target(
+            name: "TagLibWrappers",
+            dependencies: ["TagLib"],
+            path: "./Sources/TagLibWrappers"
+        ),
+        .target(
+            name: "TagLibKit",
+            dependencies: ["TagLibWrappers"]
```
[the remark this answers is not in the document]
Comment (SDGGiesbrecht) [reply Package.swift]: Unless you actually want another module of your own alongside it, just delete this and all references to it (delete the tests and point the product at `TagLib` instead, just like the code I posted before).

**Comment by SDGGiesbrecht** [comment]
That wrapper you found is exactly the shim layer I was saying you would have to write to stitch the gap between C++ and Swift. The source code cannot tell where it is importing from, so you could just as easily turn it into a package too. The wrapper package would then depend on the the package of TagLib itself.

**Comment by SDGGiesbrecht** [comment]
I would suggest wiping these two repositories and starting clean.

1. If you directly fork the [official TagLib](https://github.com/taglib/taglib) repository, it will be much easier to pull any updates. To turn it into a package, you shouldn’t only have to two files:

    1. Create a `Package.swift` (using the code I post somewhere around here basically verbatim).
    2. Copy the one header that has the `.h.cmake` file extension and remove `.cmake` from the end of the name. (Essentially adding an identical `.h` file right beside it.)

    When you make a fork, you want to make sure your changes are as clean and minimal as possible, to avoid conflicts when you try to pull the updates that have been made upstream.

2. Then create a direct fork of [TaglibWrapper](https://github.com/ryanfrancesconi/TaglibWrapper). Here you should only need to add a `Package.swift`

Don’t touch any other files unless you absolutely have to. And if you do, think about how you can isolate your changes from the rest of the project to reduce merge conflicts later on.

File: Package.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (NCrusher74) [reply Package.swift]: Okay, I think the issue I was running into was that every time I tried to delete this, the package would lose its target and wouldn't build? I hadn't realize that `.target = "TagLibKit"` was referring to this specific file and not the whole package.

I'll do what you recommend at the end of this thread, then see if I can figure out that `.target` should be.

**Comment by SDGGiesbrecht** [comment]
I got `taglib` turned into a package, but I don’t have push access to https://github.com/NCrusher74/TagLib-SPM to create a pull request.

**Pull request closed without merging**